File: apps/backend-hash/discounts/src/main/scala/com/pasviegas/discounts/database/models/Product.scala
package com.pasviegas.discounts.database.models
import com.pasviegas.discounts.database.Model
import scalikejdbc.WrappedResultSet

case class Product(id: String, title: String, description: String, priceInCents: Int) extends Model

object Product {
  def apply(resultSet: WrappedResultSet): Product = Product(
    resultSet.string("id"),
    resultSet.string("title"),
    resultSet.string("description"),
    resultSet.int("price_in_cents")
  )
}

File: apps/backend-hash/products/src/database/models/Product.js
class Product {
  static fromDatabase({ id, title, description, price_in_cents }) {
    return new Product({
      id,
      title,
      description,
      priceInCents: price_in_cents,
    });
  }

  constructor({ id, title, description, priceInCents }) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.priceInCents = priceInCents;
  }
}

module.exports.Product = Product;
